com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/flowable/FlowableFacade.java

```diff
@@ -229,13 +229,20 @@ public void deleteProcessInstance(String processInstanceId, String deleteReason)
         long overallAbortDeadline = System.currentTimeMillis() + DEFAULT_ABORT_WAIT_TIMEOUT_MS + DEFAULT_ABORT_TIMEOUT_MS;
         while (true) {
             try {
+                if (isProcessInstanceAtReceiveTask(processInstanceId)) {
```

Comment:
Do it outside the loop.

Reply:
It should remain in the outer loop, because of OptimisticLockingException mitigation